File: src/main/java/org/pyjjs/scheduler/core/api/impl/actors/resource/placing/algorithms/PlacingAlgorithm.java
package org.pyjjs.scheduler.core.api.impl.actors.resource.placing.algorithms;

import org.pyjjs.scheduler.core.api.impl.actors.resource.placing.TimeLine;

import java.util.Set;

public interface PlacingAlgorithm {

    void configurate(Set<TimeLine.TimePart> freeTimes, Set<TimeLine.UsedTime> usedTimes);

    void findPlacement(Long duration, boolean shared);



}

File: src/main/kotlin/org/pyjjs/scheduler/core/api/impl/actors/resource/placing/algorithms/PlacingAlgorithm.kt
package org.pyjjs.scheduler.core.api.impl.actors.resource.placing.algorithms

import org.pyjjs.scheduler.core.api.impl.actors.resource.placing.TimeLine

interface PlacingAlgorithm {

    fun configurate(freeTimes: Set<TimeLine.TimePart>, usedTimes: Set<TimeLine.UsedTime>)

    fun findPlacement(duration: Long?, shared: Boolean)

}
